### PR TITLE
Partially rewrite abstract and add mention of LAMMPS GUI to it

### PR DESCRIPTION
**Pull request by akohlmey**
Here is my suggestion for the abstract that adds a mention of LAMMPS GUI to the abstract.

### CODE REVIEW

**Comment by akohlmey** [comment]
I have a few questions about further changes:
- In some past projects we have use comments prefixed with initials to indicate who commented. In some other projects we used custom macros that would insert the comment in small, colored font into the article. Which is preferred? I used the initialized comment for now.
- Having long lines without wrapping makes it harder to see differences between versions. Any objections to me wrapping them as I make my way through the text?
- My next target would be to rewrite section 2.2 for using pre-compiled executables and add a section 2.3 that had a high-level introduction to LAMMPS GUI (focusing on a minimum set of features as a text editor) with a reference to the LAMMPS manual for all details. Further features (e.g. running/stopping LAMMPS, log, graphs, snapshots) are best introduced during the tutorials when the features are needed.
- I am thinking that for the purpose of the tutorials and also for the convenience of LAMMPS GUI users it would be helpful to add a "save file" shortcut button next to the "run", "stop", "create image" buttons in the bottom left of the GUI. Even though it is not necessary to save the buffer to run LAMMPS from the GUI, any crash (shouldn't happen, but beginners will find bugs where nobody looked for them before) will lose the changes. I personally prefer using keyboard shortcuts (due to my training in using emacs as text editor), but I know many folks these days prefer the buttons and calling them from the menu is getting tedious fast. Any thoughts on this change to the GUI?

I'll move on to the next branch for editing the next sections.

File: lammps-tutorials.tex [hunk 1]
```diff
@@ -103,7 +105,9 @@
...
...es and wide array of options and features as well as possible pitfalls can make it challenging to use them correctly, particularly for beginners in the topic of simulations.  LAMMPS is one such versatile molecular simulation code designed for modeling particle-based systems for a broad range of materials science and computational chemistry applications from atomistic, over coarse--grained, mesoscale, and grid-free continuum models to discrete element modeling.  LAMMPS is capable of efficiently running simulations of varying sizes from small desktop computer to large-scale supercomputing environments. Its flexibility and extensibility makes it ideal for complex and extensive simulations of atomic and molecular systems and beyond.  This article introduces a suite of tutorials designed to make learning LAMMPS more accessible to new users.  The first four tutorials cover the basics of running molecular simulations in LAMMPS with systems of varying complexities, including a simple fluid and a carbon nanotube.  The last three tutorials address more advanced molecular simulation techniques, specifically the use of a reactive force field, enhanced sampling, and grand canonical Monte Carlo.
```
Comment (jrgissing) [inline lammps-tutorials.tex:108]: suggest rephrasing to ...." chemistry applications, including atomistic, coarse--grained, mesoscale, and grid-free continuum models."

**Comment by akohlmey** [comment]
Another general issue related to the LAMMPS GUI. On Windows and macOS the file extension .lmp is registered to be linked with the LAMMPS GUI executable. For "make install" on Linux there is also an XML file that will be installed to `${CMAKE_INSTALL_DATADIR}/mime/text/x-application-lammps.xml` and should also register the .lmp extension with the LAMMPS GUI executable on most Linux desktop environments, but that has been hit-or-miss so far. If I ever get around to making a flatpak package of the GUI for Linux, that would probably work out of the box.

Thus it may be convenient to also use the .lmp extension

File: lammps-tutorials.tex (hunk 1 above)
Comment (akohlmey) [reply lammps-tutorials.tex]: Thanks, done.

**Comment by simongravelle** [comment]
> Thus it may be convenient to also use the .lmp extension

Noted, fine by me.

File: lammps-tutorials.tex [hunk 2]
```diff
@@ -103,7 +105,9 @@
...
...are packages and wide array of options and features as well as possible pitfalls can make it challenging to use them correctly, particularly for beginners in the topic of simulations.  LAMMPS is one such versatile molecular simulation code designed for modeling particle-based systems for a broad range of materials science and computational chemistry applications, including atomistic, coarse--grained, mesoscale, grid-free continuum, and discrete element models.  LAMMPS is capable of efficiently running simulations of varying sizes from small desktop computer to large-scale supercomputing environments. Its flexibility and extensibility makes it ideal for complex and extensive simulations of atomic and molecular systems and beyond.  This article introduces a suite of tutorials designed to make learning LAMMPS more accessible to new users.  The first four tutorials cover the basics of running molecular simulations in LAMMPS with systems of varying complexities, including a simple fluid and a carbon nanotube.  The last three tutorials address more advanced molecular simulation techniques, specifically the use of a reactive force field, enhanced sampling, and grand canonical Monte Carlo.
```
Comment (simongravelle) [inline lammps-tutorials.tex:108]: two suggestions:

Its flexibility and extensibility makes it -> Its flexibility and extensibility **make** it

from small desktop computer to large-scale supercomputing environments --> from small desktop **computers** to large-scale supercomputing environments

**Review by simongravelle — APPROVED** [review]

**Comment by simongravelle** [comment]
> * In some past projects we have use comments prefixed with initials to indicate who commented. In some other projects we used custom macros that would insert the comment in small, colored font into the article. Which is preferred? I used the initialized comment for now.

I let you choose, I have very little experience of writing a latex document collaboratively (The first, simpler, option sounds good enough, though).

**Comment by simongravelle** [comment]
> * Having long lines without wrapping makes it harder to see differences between versions. Any objections to me wrapping them as I make my way through the text?

No objection. You are right, this would be much better to wrap the line here.

**Comment by simongravelle** [comment]
> * I also noticed that the text files contain DOS/Windows style CR/LF line endings which should not be used when importing text on Linux. My latex version complains heavily about the unexpected characters. A quick treatment with dos2unix should get rid of those.

I see. This is an issue I did not even know I could have. :)

Did you fix it already? Because when I run dos2unix to the tex file (in the main branch or in here), git does not detect any changes.

Also, it seems that my text editor is already set to LF by default... @jrgissing are you using windows, may be?

File: lammps-tutorials.tex (hunk 2 above)
Comment (akohlmey) [reply lammps-tutorials.tex]: Done. Thanks.

**Comment by akohlmey** [comment]
> Did you fix it already? Because when I run dos2unix to the tex file (in the main branch or in here), git does not detect any changes.

Same for me. Perhaps I am misinterpreting the warnings. I get a lot of these two:
```
LaTeX Warning: Command \r invalid in math mode on input line 2192.
LaTeX Warning: Characters dropped after `\end{verbatim}' on input line 2207.
```

**Comment by akohlmey** [comment]
@simongravelle unless you see anything else that needs changing, I am done with this branch. I'll continue with my LAMMPS+GUI updates (still need to port the library interface changes to fortran and add tests and documentation integration, and also update the Howto documentation for LAMMPS GUI). High time to get some sleep.

**Comment by simongravelle** [comment]
> * Any thoughts on this change to the GUI?

So, in case of crash, the user could recover the current state of the simulation (log, graph)? And there would be some type of hidden binary file saved within the same folder as the input that the user would not see?

I don't have a strong opinion. As it is now, I like that one can switch from the GUI to a classic text editor and there is no difference and no extra file getting saved in the folder. However, I suppose that if the saving and reloading is done manually and not done automatically every time the GUI is used, then it could indeed be useful without being anoying.

**Review by simongravelle — APPROVED** [review]

**Comment by akohlmey** [comment]
> So, in case of crash, the user could recover the current state of the simulation (log, graph)? And there would be some type of hidden binary file saved within the same folder as the input that the user would not see?

The LAMMPS library interface is set up in such a way that (most) LAMMPS errors result in exceptions that are caught and can be checked for through the library interface. Thus when there is an error in the input and LAMMPS crashes, the GUI just needs to issue a "clear" command and LAMMPS can start from scratch. In fact, the "clear" is automatically inserted every time you hit the run command or button.

So this is only about crashes of the GUI (which *should* not happen) Since I have seen limited feedback on it, there are likely a whole bunch of bugs looming when people do random stuff (like beginners tend to do). I think if we want to support people that want to use LAMMPS GUI just as a custom editor for LAMMPS inputs and run on the console, then fast access to "save buffer" from the GUI (and not just form hitting CTRL-S) is an improvement for beginners.

It was easy enough to add. Just check out the new snapshot package and compare to the old version.

**Pull request merged**